packages/ui-components/src/components/Editor/SqlEditor.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState } from 'react';
+import parser from 'js-sql-parser';
+import BaseAceEditor from 'react-ace';
+import styled from 'styled-components';
+import 'ace-builds/src-noconflict/mode-pgsql';
+import 'ace-builds/src-noconflict/theme-xcode';
+import 'ace-builds/src-noconflict/ext-language_tools';
+import type { Ace } from 'ace-builds';
+import { GREY_DE } from '../../../stories/story-utils/theme/colors';
+
+const editorName = 'sqlEditor';
+
+const AceEditor = styled(BaseAceEditor).attrs({
+  maxLines: 20,
+  minLines: 6,
+  showPrintMargin: false,
+  tabSize: 2,
+  theme: 'github',
+  width: '100%',
+  setOptions: {
+    enableLiveAutocompletion: true,
+    enableBasicAutocompletion: true,
+  },
+})`
+  border: 1px solid ${({ theme }) => theme.palette?.grey?.['400'] ?? GREY_DE};
```

Comment:
I sorta expected `${({ theme }) => theme.palette.grey.['400’]}` to work, but I presume this is failing me because `admin-panel` doesn’t seem to have a theme defined?

Is this cascade/fallback is appropriate?

Reply:
It definitely should have one defined - when you say it's failing you, do you mean it doesn't show up at all when viewing in the admin panel?

Reply:
It tells me that `grey` is undefined and crashes—the entire web app goes blank

Reply:
Oh I see - this is because the VizBuilder, while being technically part of the admin panel, is actually it's own app, which doesn't have a theme. In this case I think it's fine to use some hard coded colours (or create colors constants for this file) because it'll only ever be used in the viz builder. Alternatively you can set the viz builder up to have a theme provider (check out the main admin panel `main.jsx` for an example)

Reply:
Ah, that’s definitely useful to know—thank you!